db/memorystore.go

```diff
@@ -322,3 +323,32 @@ func keyToID(key crypto.PublicKey) (string, error) {
 		return hex.EncodeToString(spkiDigest[:]), nil
 	}
 }
+
+// GetCertificateBySerial loops over all certificates to find the one that matches the provided
+// serial number. This method is linear and it's not optimized to give you a quick response.
+func (m *MemoryStore) GetCertificateBySerial(serialNumber *big.Int) *core.Certificate {
+	m.RLock()
+	defer m.RUnlock()
+	for _, c := range m.certificatesByID {
+		if serialNumber.Cmp(c.Cert.SerialNumber) == 0 {
+			return c
+		}
+	}
+
+	return nil
+}
+
+// GetCertificateBySerial loops over all revoked certificates to find the one that matches the
```

Comment:
typo: the function name in your Godoc comment here is the wrong one :-)
```suggestion
// GetRevokedCertificateBySerial loops over all revoked certificates to find the one that matches the
```

Reply:
Thanks! Updated.